src/Serilog/Parameters/PropertyValueConverter.cs

```diff
@@ -154,9 +156,16 @@ LogEventPropertyValue CreatePropertyValue(object value, Destructuring destructur
 
             if (destructuring == Destructuring.Destructure)
             {
-                var typeTag = value.GetType().Name;
-                if (typeTag.Length <= 0 || !char.IsLetter(typeTag[0]))
+                var type = value.GetType();
+                var typeTag = type.Name;
+                var typeInfo = type.GetTypeInfo();
+                // Check if the type is an anonymous type
+                if (typeTag.Length <= 0
```

Comment:
I'd be tempted to move all the rest of the condition here into the `IsCompilerGeneratedTypeName` function, since it is going to be inlined anyway - otherwise it's a bit tricky for example to trace that the `Length > 0` precondition that function has is met without skipping back up to the call site.